cmd/ssh.go

```diff
@@ -222,6 +223,90 @@ func copySourceAndUnzip(execID, rootDir, dstPath string, connectionInfo types.Co
 	return nil
 }
 
+func copyDirFromRemoteAndUnzip(sshTarget, localDirectory, privateKey string) error {
+	ui.Infof("🧳 Gathering context from %q", sshTarget)
+
+	remotePath, err := zipRemoteDirectory(sshTarget, privateKey)
+	if err != nil {
+		return fmt.Errorf("Failed to zip the remote directory. Expected both a remote target and directory in %s", sshTarget)
+	}
+
+	ui.Infof("📦 Copying the archive of the remote path to the host...")
+
+	sshTargetAndDir := strings.Split(sshTarget, ":")
+	if len(sshTargetAndDir) != 2 {
+		return fmt.Errorf("Expected target to be in the format 'user@host:directory'")
+	}
+	sshTargetDirectory := sshTargetAndDir[0] + ":" + remotePath
+	remoteFilename := filepath.Base(remotePath)
+	archiveLocalTargetDir := config.GetGlobalConfigPath()
+	archiveLocalTarget := filepath.Join(archiveLocalTargetDir, remoteFilename)
+
+	err = copySourceSCP(sshTargetDirectory, config.GetGlobalConfigPath(), privateKey)
+	if err != nil {
+		return fmt.Errorf("Failed to copy the archive of your remote path to the host. "+
+			"Please check if %s exists on the remote and ensure Unweave has the necessary permissions to access %s",
+			sshTargetDirectory, archiveLocalTargetDir)
+	}
+
+	ui.Infof("🗜️ Unzipping the copied archive to the local directory...")
+
+	cmd := exec.Command("tar", "-xf", archiveLocalTarget, "-C", localDirectory, "--strip-components=1")
+
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("Failed to unzip the copied archive of your code from %s to %s. "+
+			"Please ensure that Unweave has the necessary permissions to access %s and perform this operation manually",
+			archiveLocalTarget, localDirectory, archiveLocalTarget)
+	}
+
+	return nil
+}
+
+// zipRemoteDirectory takes an ssh target, and zips up the contents of that target to a returned in the remote /tmp
+func zipRemoteDirectory(sshTarget, privateKeyPath string) (string, error) {
```

Comment:
I suggest adding named return values here for clarity

Reply:
i don't think that is too unclear as it's hinted at in the docstring, did you have trouble identifying that the remote directory path was the return string here?

Reply:
Updated